src/Core/Domain/Product/Combination/Command/UpdateCombinationStockCommand.php

```diff
@@ -0,0 +1,234 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command;
+
+use DateTimeInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+
+/**
+ * Updates combination stock information
+ */
+class UpdateCombinationStockCommand
+{
+    /**
+     * @var CombinationId
+     */
+    private $combinationId;
+
+    /**
+     * @var int|null
+     */
+    private $quantity;
+
+    /**
+     * @var int|null
+     */
+    private $minimalQuantity;
+
+    /**
+     * @var string|null
+     */
+    private $location;
+
+    /**
+     * @var int|null
+     */
+    private $lowStockThreshold;
+
+    /**
+     * @var bool|null
+     */
+    private $lowStockAlertOn;
+
+    /**
+     * @var DateTimeInterface|null
+     */
+    private $availableDate;
+
+    /**
+     * @var bool
+     */
+    private $movementOn = true;
+
+    /**
+     * @param int $combinationId
+     */
+    public function __construct(
+        int $combinationId
+    ) {
+        $this->combinationId = new CombinationId($combinationId);
+    }
+
+    /**
+     * @return CombinationId
+     */
+    public function getCombinationId(): CombinationId
+    {
+        return $this->combinationId;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getQuantity(): ?int
+    {
+        return $this->quantity;
+    }
+
+    /**
+     * @param int $quantity
+     *
+     * @return UpdateCombinationStockCommand
+     */
+    public function setQuantity(int $quantity): UpdateCombinationStockCommand
+    {
+        $this->quantity = $quantity;
+
+        return $this;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getMinimalQuantity(): ?int
+    {
+        return $this->minimalQuantity;
+    }
+
+    /**
+     * @param int $minimalQuantity
+     *
+     * @return UpdateCombinationStockCommand
+     */
+    public function setMinimalQuantity(int $minimalQuantity): UpdateCombinationStockCommand
+    {
+        $this->minimalQuantity = $minimalQuantity;
+
+        return $this;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getLocation(): ?string
+    {
+        return $this->location;
+    }
+
+    /**
+     * @param string $location
+     *
+     * @return UpdateCombinationStockCommand
+     */
+    public function setLocation(string $location): UpdateCombinationStockCommand
+    {
+        //@todo: validate with regex pattern
+        $this->location = $location;
```

Comment:
I don't think we have a pattern for this since any warehouse can have different patterns, it's just limited in size

Reply:
oh, now its getting interesting :sweat_smile:
* Combination $location is validated by `isGenericName`
* Product $location is validated by `isReference`
* StockAvailable $location is validated by `isString` 
:sweat: 

I wonder if we simply could use the one from StockAvailable, so it would be just a simple string with length of 255 for all of them? :thinking:

Reply:
I homogenized them according to StockAvaialble (isString 255)
However in SQL the stock_available location DEFAULT was empty string, while combination&product had NULL, so I also made stock_available DEFAULT NULL.